### PR TITLE
add refined logging to the system

### PR DESCRIPTION
**Pull request by wx1988**
change the log interface to enable more information to be logged. 

In the javascript, add detail log at key user interactions.

### CODE REVIEW

**Comment by wx1988** [comment]
Missing function: 
The region selection is not logged.

File: static/js/showDetailTable.js [hunk 1]
```diff
@@ -1,19 +1,23 @@
...
  * */
 
-// TODO, rename
 detailViewData = {
   lastSelectCaseno: undefined,
   table: undefined,  // google table view
-  data : undefined   // google data 
+  data : undefined,   // google data 
+  acc_column_list : ['Caseno', 'date', 'time', 'road surface','Alcohol Flag', 'Light', 'Driver Gender', 'Location Type','Driver age', 'NumK', 'NumA', 'NumB', 'NumC'],
+  acc_type_list : ['number', 'string', 'string', 'string', 'string', 'string', 'string', 'string', 'string', 'number', 'number', 'number', 'number'],
```
Comment (singhsume123) [inline static/js/showDetailTable.js]: Are these string. Or date and time type? In view mode its showing me correct format. Ignore this old commit

File: templates/index.html [hunk 2]
```diff
@@ -164,28 +171,29 @@ <h4 class="panel-title">
...
                     <div id="userinfo"></div>
                 </div>
-                <div class="col-md-6">
+                <div class="col-md-4">	
+                    <button class="btn btn-success" id="details">
+                        View Details</button> 
+                    <button class="btn btn-success" id="export">Export Data</button>
+                </div>
+                <div class="col-md-5">
                     <button class="btn btn-success btn-pressure pull-right" id="roadsegments">RoadSegments</button>
```
Comment (singhsume123) [inline templates/index.html:191]: Why it has been moved at the top?

Comment (wx1988) [inline templates/index.html:191]: When all the facet filtering panel are expanded, these buttons will be out of screen.

File: code.py [hunk 3]
```diff
@@ -45,83 +74,96 @@ def get_current_user_id():
...
+            data = get_accidents_api(facet_obj)
+            print 'accident number', len(data)
+            #return simplejson.dumps(res, )
+            return simplejson.dumps({
+                'status':0,
+                'data':data},
+                ignore_nan=True)
+        except Exception as e:
+            return json.dumps({
+                'status':0,
```
Comment (singhsume123) [inline code.py]: Should it be 1 or 0?

File: code.py [hunk 4]
```diff
@@ -45,83 +74,96 @@ def get_current_user_id():
...
+        try:
+            from db_api import create_log
+            d = web.input()
+            log_info = json.loads( d['info'] )
+            log_info["user_id"] = get_current_user_id()
+            create_log(log_info)
+            return json.dumps({'status':0})
+        except Exception as e:
+            return json.dumps({
+                'status':0,
```
Comment (singhsume123) [inline code.py]: same as above

File: static/js/common_util.js [hunk 5]
```diff
@@ -6,88 +6,101 @@ if (!String.format) {
...
+    return null;
+  }
+  else{
+    return results[1] || 0;
+  }
+}
+
+function getMapBound(map){
+  var bound = map.getBounds();
+  if(bound == undefined)return undefined;
```
Comment (singhsume123) [inline static/js/common_util.js]: better add brackets to add clarity.

File: code.py (hunk 3 above)
Comment (wx1988) [inline code.py]: should be 1, fixing now.

File: static/js/common_util.js (hunk 5 above)
Comment (wx1988) [inline static/js/common_util.js]: fixed.

File: templates/index.html [hunk 6]
```diff
@@ -197,7 +199,15 @@ <h4 class="panel-title">
                 <div id="map"></div>
             </div>
             <div class="row" id ="table_row">
-                <div id="table_div"></div>
+                <div id="table_div" class="col-md-8"></div>
+                <div id="visual_div" class="col-md-4">
+                    <select id='chartSelect'>
+                        <option value="severity">Severity</option>
```
Comment (singhsume123) [inline templates/index.html:205]: has the idea of hovering been dropped?

File: static/js/drawFigure.js [hunk 7]
```diff
@@ -0,0 +1,102 @@
...
+        event2num[ acc_list[i].events[j] ] =0;
+      event2num[ acc_list[i].events[j] ] += 1;
+    }
+  }
+  // sort
+  var tmpData = []
+  for( var k in event2num ){
+    tmpData.push([accCode.event[k], event2num[k]]);
+  }
+  tmpData.sort( function(a,b){return b[1] - a[1];})
```
Comment (singhsume123) [inline static/js/drawFigure.js]: remove extra space.

Comment (singhsume123) [inline static/js/drawFigure.js]: remove extra space.

File: static/js/drawFigure.js [hunk 8]
```diff
@@ -0,0 +1,102 @@
...
+  }
+  var numk = 0;
+  var numa = 0;
+  var numb = 0;
+  var numc = 0;
+  for(var i = 0;i < acc_list.length;i++){
+    if(acc_list[i].num_k > 0) numk += 1;
+    if(acc_list[i].num_a > 0) numa += 1;
+    if(acc_list[i].num_b > 0) numb += 1;
+    if(acc_list[i].num_c > 0) numc += 1;
```
Comment (singhsume123) [inline static/js/drawFigure.js:45]: better make it if-elseif. I hope it would be just one type?

File: static/js/drawFigure.js [hunk 9]
```diff
@@ -0,0 +1,102 @@
...
+  var tmpData = [ ['Severity', 'Number'] ];
+  tmpData.push( ['K', numk] );
+  tmpData.push( ['A', numa] );
+  tmpData.push( ['B', numb] );
+  tmpData.push( ['C', numc] );
+
+  tmpData = google.visualization.arrayToDataTable(tmpData);
+  var options = { title: 'Severity Statistics' };
+  var chart = new google.visualization.PieChart( document.getElementById('googleChartDiv'));
+  chart.draw(tmpData, options );
```
Comment (singhsume123) [inline static/js/drawFigure.js]: Remove extra space.

File: static/js/drawFigure.js [hunk 10]
```diff
@@ -0,0 +1,102 @@
...
+  } 
+  var tmpData = [ ['Severity', 'Number'] ];
+  tmpData.push( ['K', numk] );
+  tmpData.push( ['A', numa] );
+  tmpData.push( ['B', numb] );
+  tmpData.push( ['C', numc] );
+
+  tmpData = google.visualization.arrayToDataTable(tmpData);
+  var options = { title: 'Severity Statistics' };
+  var chart = new google.visualization.PieChart( document.getElementById('googleChartDiv'));
```
Comment (singhsume123) [inline static/js/drawFigure.js]: remove extra space.

File: templates/index.html (hunk 6 above)
Comment (wx1988) [inline templates/index.html:205]: This is a temporary solution.

File: static/js/drawFigure.js (hunk 8 above)
Comment (wx1988) [inline static/js/drawFigure.js:45]: Each accident can be assigned to multiple categories.

**Pull request merged**